demo/app.component.ts

```diff
@@ -1,4 +1,4 @@
-import { Component, OnInit } from '@angular/core';
```

Comment:
I'm not sure I understand your comment.
You don't want to use ViewChild to get the controllers?